### ISSUE TITLE
Allow grammars to exclude themselves in menu

### ISSUE THREAD

**Issue by Alhadis** [atom/grammar-selector#30]
Some grammars shouldn't have entries in the selection menu. For instance, the [Hyperlink language](https://github.com/atom/language-hyperlink) only exists to inject those helpful underlines on any URL-like sequence in the editor, irrespective of language:

<img width="745" src="https://cloud.githubusercontent.com/assets/2346707/14233816/48c0014e-fa16-11e5-9a3f-9aa32e0c1e80.png" />

Other grammars exist on their own to be loaded by others (e.g., "[Regular Expression (Javascript)](https://github.com/atom/language-javascript/blob/master/grammars/regular%20expressions%20(javascript).cson)".

In short, it'd be great if grammars could flag themselves as hidden. Possibly by having a property set in the grammar file?
```coffee
'name': 'Hyperlink'
'scopeName': 'text.hyperlink'
'fileTypes': []
'hideInMenu': true
```

It's tempting to rely on an empty `fileTypes` array to exclude the grammar instead, but that would assume the grammar's author *intended* for a user to never select it from the file-menu.

**Comment by Alhadis** [comment]
FYI, I've started looking at this locally (as the filename in the image would suggest), but I'm unable to get my development copy of `first-mate` running. Calls to `console.log` are showing up normally, but not if they're running from inside the Grammar class's constructor. I'm assuming the grammars are being instantiated externally (outside the running Atom window)...
